src/core/router/index.js

```diff
@@ -34,11 +34,7 @@ router.beforeResolve((to, from, next) => {
   } else {
     if (store.state.wallet.address === null) {
       store.dispatch('external/setLastPath', to.path);
-      /** only resolve a new path when current path isn't accesswallet
-       * this is a little hacky but utilizes intended purpose.
-       * vue router notes: move on to the next hook in the pipeline. If no hooks are left, the navigation is confirmed.
-       */
-      if (from.name !== 'AccessWallet') router.push({ name: 'AccessWallet' });
+      next({ name: 'AccessWallet' });
```

Comment:
merge the routes branch with this and replace `AccessWallet` with the right route